screensaverd.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Simple screen saver daemon. Turns off the display after idle timeout.
+ * Turns the display on on any event from /dev/input/event* files.
+ * On exit turns the display on.
+ *
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Igor Zhbanov <igor.zhbanov@jolla.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define _BSD_SOURCE
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <dirent.h>
+#include <signal.h>
+#include <stdarg.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <sys/select.h>
+
+#define UNUSED			__attribute__((unused))
+
+#define DISPLAY_CONTROL		"/sys/class/graphics/fb0/blank"
```

Comment:
I thought about it. Variants are:
1) Dozen of #ifdef/#else/#endif
2) Separate *.h for each configuration used as a parameter to make in RPM-spec
3) Command-line parameter provided by recovery menu
4) Super-intelligent application traversing sysfs and looking for suitable file from a big list
5) Your variant

Please vote. ;-)